Enabling log streaming

marians opened the pull request:
Ping @svendvd @stephanpavlovic - This change introduces the `streamLogs` function with an example given in the README.

ewoutp commented on README.md:
```diff
@@ -28,6 +24,33 @@ GiantSwarm.applicationStatus("my-org", "my-env", "my-app", function(d){
+}
+
+var organizationName = 'my-org';
+var instanceIds = ['4lyqvwvqhg0m'];
```
I guess you don't want this "constant" in here.

marians replied:
I don't mind having it in the README as an example.

ewoutp commented on lib/client.js:
```diff
@@ -172,6 +172,31 @@ var GiantSwarm = (function () {
+      var uri = '/v1/org/' + organizationName + '/stream/logs';
+      request
+        .post(apiEndpoint + uri)
+        .set('Authorization', 'giantswarm ' + authToken)
```
Suggest to make a function to set these headers.

marians replied:
@ewoutp Can you elaborate a bit? I don't understand what you mean.

ewoutp replied:
The authorization header (and probably a few others) will have to be set in all API calls.
So I suggest to group them in a separate function.

marians commented:
@ewoutp I added the `_authorizedRequest()` method to set the `Authorized` header.

Also added basic testing. Will write more test cases after merge.

ewoutp commented on lib/client.js:
```diff
@@ -8,6 +8,12 @@ var GiantSwarm = (function () {
+      if (authToken) {
+        return request.set('Authorization', 'giantswarm ' + authToken);
+      }
+    },
```
Shouldn't there be an else (`return request`)?

marians replied:
There should definitely be an `else`, but I'd rather use it to throw an exception. I'd like to add this after merge with PR #3, since there is a already a useful exception defined in that branch.

ewoutp replied:
ok

ewoutp commented on lib/client.js:
```diff
@@ -172,11 +176,39 @@ var GiantSwarm = (function () {
+            errorCallback(err);
+          } else {
+            if (resp.body.status_code != 10000) {
+              errorCallback();
```
Shouldn't we pass `resp.body` here?

marians replied:
pass to where?

ewoutp replied:
`errorCallback(resp.body);`

zyndiecate commented on spec/client_spec.js:
```diff
@@ -0,0 +1,27 @@
+        expect(value).toEqual(1);
+        done();
+      });
+    });
```
It would probably be better to support the error callback here as well, to have a clean error just in case. Otherwise this test will cause yasmine to only produce a timeout error after 5 seconds. This would slow the whole testsuite down when things go terribly wrong.

zyndiecate commented on spec/client_spec.js:
```diff
@@ -0,0 +1,27 @@
+
+    it("should not allow me to ping the server", function(done){
+      GiantSwarm.setApiEndpoint('https://www.google.com');
+      GiantSwarm.ping(function(){}, function(err){
```
Same here as above. The success callback should raise an error.

marians commented:
@zyndiecate Very good, thanks! I changed the two tests.

marians commented:
I know these aren't really LGTMs here. Will try to address the rest of @ewoutp's comments in #3 and merge this here now. Otherwise things will be too complicated for me.
